vendor/sigs.k8s.io/controller-runtime/pkg/controller/controllerutil/controllerutil.go

```diff
@@ -387,6 +390,57 @@ func ContainsFinalizer(o client.Object, finalizer string) bool {
 	return false
 }
 
+// RemoveDeletedNodeFinalizer removes finalizers from the object for deleted nodes.
+func RemoveDeletedNodeFinalizer(c client.Client, o client.Object) (error, string) {
```

Comment:
Please don't patch files under `vendor/`, that should be an exception (e.g. if we need to bring in a patch that hasn't been released or merged yet)